### DISCUSSION TITLE
Health Factor Check

### DISCUSSION THREAD

**Discussion by maheshbsl** [Cyfrin/foundry-full-course-cu#2633] [General]
This is the code we are using to calculate health factor

```
function _calculateHealthFactor(
        uint256 totalDscMinted,
        uint256 collateralValueInUsd
    )
        internal
        pure
        returns (uint256)
    {
        if (totalDscMinted == 0) return type(uint256).max;
        uint256 collateralAdjustedForThreshold = (collateralValueInUsd * LIQUIDATION_THRESHOLD) / LIQUIDATION_PRECISION;
        return (collateralAdjustedForThreshold * PRECISION) / totalDscMinted;
    }
    ````
    In this function, I don't get why we are dividing the collateralAdjustedForThreshold  which is usd value by totalDscMinted wich is ether value.

**Comment by EngrPips** [comment]
Hello @maheshbsl, `collateralAdjustedForThreshold` and `totalDscMinted` are both in `USD` value. The protocol is designed to have a standard for representing `1 USD`, which might be `1 ether = 1 USD` check the contract implementation carefully to see this.

**Reply by SwiftDrip** [reply]
Awesome post! Thanks for sharing the knowledge and keeping up the good work. 

For more details, you can visit **[SwiftDrip](https://swiftdrip.com)**.

**Reply by EngrPips** [reply]
Hmmmmm 🤔

**Reply by maheshbsl** [reply]
Thank you bro, I got it now,  we pegged `1 dsc = 1 usd` .

**Reply by EngrPips** [reply]
Anytime, Friend. I am glad you get the gist.

**Reply by maheshbsl** [reply]
Is writing tests the hardest part of smart contracts development?

**Reply by EngrPips** [reply]
I believe every part is equally hard.
